packages/components/src/Popover/Popover.tsx

```diff
@@ -0,0 +1,109 @@
+/* eslint-disable no-null/no-null */
+import React, { useState } from "react";
+import { usePopper } from "react-popper";
+import classnames from "classnames";
+import classes from "./Popover.css";
+import { ButtonDismiss } from "../ButtonDismiss";
+
+export interface PopoverProps {
+  /**
+   * Element the popover will attach to and point at.
+   */
+  readonly attachTo: Element | React.RefObject<Element | null>;
+
+  /**
+   * Pop-over content.
+   */
+  readonly children: React.ReactNode;
+
+  /**
+   * Whether or not the popover is dismissable via the header close button.
+   */
+  readonly dismissible?: boolean;
+
+  /**
+   * Control popover viability.
+   */
+  readonly open?: boolean;
+
+  /**
+   * Callback executed when the user wants to close/dismiss the popover
+   */
+  readonly onRequestClose?: () => void;
+
+  /**
+   * Describes the preferred placement of the popper.  The first element in the placement array will be taken
+   * as the preferred location with the rest of the array as fallback.
+   */
+  readonly preferredPlacement?: "top" | "bottom" | "left" | "right" | "auto";
+}
+
+export function Popover({
+  onRequestClose,
+  children,
+  dismissible = true,
+  attachTo,
+  open,
+  preferredPlacement = "auto",
+}: PopoverProps) {
+  const [popperElement, setPopperElement] = useState<HTMLElement | null>(null);
+  const [arrowElement, setArrowElement] = useState<HTMLElement | null>(null);
+  const { styles: popperStyles, attributes } = usePopper(
+    attachTo instanceof Element ? attachTo : attachTo.current,
+    popperElement,
+    {
+      modifiers: buildModifiers(arrowElement),
+      placement: preferredPlacement,
+    },
+  );
+
+  const popperClassName = classnames(classes.popover);
+  const arrowClassName = classnames(classes.arrow);
+  const dismissButton = classnames(classes.dismissButton);
+
+  return (
+    open && (
+      <div
+        data-testid="popover"
```

Comment:
It would be nice to have or tests check for content or something else that is visible to the user. If we do this then we shouldn't need a `data-testid` here.

Point 2 here: https://testing-library.com/docs/guiding-principles 😄